fhttp/http_test.go

```diff
@@ -941,6 +941,31 @@ func TestChangeMaxPayloadSize(t *testing.T) {
 	}
 }
 
+func TestAddHTTPS(t *testing.T) {
+	var tests = []struct {
+		input    string
+		expected string
+	}{
+		{"foo.com", "https://foo.com"},
```

Comment:
add a test for `foo` (shorter than prefix) and maybe `http` which should turn into https://http (if a host name was http)

Reply:
I might miss understand your comment here about shorter than prefix :).